docs/reference/gps.md

```diff
@@ -169,9 +169,9 @@ The returned vector indicates the absolute position of the [GPS](#gps) device.
 This position can either be expressed in the cartesian coordinate system of Webots or using latitude-longitude-altitude, depending on the value of the `gpsCoordinateSystem` field of the [WorldInfo](worldinfo.md) node.
 The `gpsReference` field of the [WorldInfo](worldinfo.md) node can be used to define the reference point of the GPS.
 
-The `wb_gps_get_speed` function returns the current [GPS](#gps) speed in meters per second.
+The `wb_gps_get_speed` function returns the current [GPS](#gps) speed in meters per second. If there is no physic node on the robot, the first returned value will be NAN.
```

Comment:
Note that the sentence `If there is no physics node on the robot` is wrong doesn't match the code.
This new code only checks if the closest parent Solid has physics not if an ancestor or top Robot node has physics.

Reply:
and what is the correct way to do it ? I would think that having an ancestor or top Robot node with physics is enough, it don't need to be the direct parent. I'm right ? or is the code correct and the documentation to be changed ?

Reply:
The code, i.e. checking only for the closest Solid, is consistent with the Accelerometer device so it would be fine to keep it like this if it is properly documented.
To change the code it should be checked if it is possibile to retrieve the speed from ODE in all the case where an ancestor node has physics, including cases where there are intermediate joint nodes.

Reply:
ok, change the doc from "robot" to "parent node" in the "If there is no physic node on the robot, the first returned value will be NAN." sentence

Reply:
i let you resolve once it's good for you